dagmc/dagnav.py

```diff
@@ -289,8 +310,17 @@ def num_triangles(self):
     def _get_triangle_sets(self):
         return [s.handle for s in self.get_surfaces().values()]
 
+
 class Group(DAGSet):
 
+    _category: str = 'Group'
+    _geom_dimension: int = 4
+
+    def __init__(self, model: DAGModel, handle: np.uint64):
+        super().__init__(model, handle)
+        if self.geom_dimension != self._geom_dimension:
+            warn(f'DAGMC group with global ID {self.id} does not have geom_dimension=4.')
```

Comment:
From slack regarding tagging of group sets with a geometry dimension value of 4: 

> Or it was part of the spec... It's in many of our older files but not in the newer ones. Means we should rely on the CATEGORY TAG entirely I'd say. I spent some time looking through old versions (like, 10+ yrs) trying to figure out when this went away, but I couldn't track it down. I'll let you know if I find anything more specific.

So we can remove this check and just check the category tag value. Thought that doesn't seem to be checked here?

Reply:
Ah right, good point. I've just updated the check to only look at the category tag. My only remaining question then is -- when we create a new group, should we be assigning geom_dimension=4 or just omit it at this point?

Reply:
Good question. I think I'd opt to assign it if it isn't there already? Perhaps with some kind of warning. It's not great but I think it'd be better to update the files to match the spec when possible.

Reply:
Oh, I was asking specifically about new groups with the thought that we would leave existing ones alone. It would seem a little surprising to me if you loaded a file and then wrote it back to disk and it changed in the process, unless like you said a warning was shown. At that point, the warning will be shown just as often as a check for no `geom_dimension` in the first place.

Reply:
I'm not worried about the warnings necessarily, more that the internal mesh instance meets the file spet.

But you're right that it's probably not a good idea to modify the mesh data behind the scenes... I guess I'd opt to set it and present a warning. At the most general level, here's what I'd do:

1. If the set handle passed in to `__init__` has only one of the `CATEGORY` or `GEOM_DIMENSION` tags set, set the other appropriately and present a warning.
2. If either those tags are set and the value is inconsistent with the class attributes or each other, raise an error.

Given that we rely on both tags in all of the classes derived from `DAGSet`, the checks should all be able to occur in it's `__init__` method to reduce a bunch of boilerplate code in the `Volume`, `Surface`, and `Group` sets.

Reply:
Sounds good -- I'll implement all that once I get a chance.

Reply:
Finally got around to this. It ended up being a little more complicated than I initially thought but seems to work. Let me know what you think!